From 8f1fb8d7dd51e1a3ec2de71e70a88886d28c8bd3 Mon Sep 17 00:00:00 2001
From: Lucius Hu <lebensterben@users.noreply.github.com>
Date: Thu, 22 Aug 2019 04:01:00 -0400
Subject: [PATCH] Specifically check the name of the shell

This is mainly for compatibility. Currently, this script is verified to work
with bash, zsh, and sh if it's aliased to bash.
---
 10-command-not-found.sh | 43 ++++++++++++++++++++++++-----------------
 1 file changed, 25 insertions(+), 18 deletions(-)

diff --git a/10-command-not-found.sh b/10-command-not-found.sh
index ac0d22d..60af07d 100644
--- a/10-command-not-found.sh
+++ b/10-command-not-found.sh
@@ -1,27 +1,34 @@
 if [ "$(expr $- : '.*i')" -ne 0 ]; then
-  command_not_found_handle () {
-    echo "$1: command not found"
-    if [ "$(id -u)" = "0" ]; then
+
+  # Get realname of the shell
+  REALNAME=$(basename "$(realpath "$(command -v "$0")")")
+
+  if [ "$REALNAME" = bash ] || [ "$REALNAME" = zsh ]; then
+
+    command_not_found_handle () {
+      echo "$1: command not found"
+      if [ "$(id -u)" = "0" ]; then
         MSG='$1 == cmd { print "To install " $1 " use:  swupd bundle-add " $2; exit }'
-    elif [ "$1" != "sudo" ] && groups | grep -q -w -e wheel -e wheelnopw ; then
+      elif [ "$1" != "sudo" ] && groups | grep -q -w -e wheel -e wheelnopw ; then
         MSG='$1 == cmd { print "To install " $1 " use:  sudo swupd bundle-add " $2; exit }'
-    else
+      else
         MSG='$1 == cmd { print "To install " $1 " your system administrator needs to do:  swupd bundle-add " $2; exit }'
-    fi
-    R=$(awk -F"\t" -v cmd="$1" "$MSG" /usr/share/clear/commandlist.csv)
-    if [ -n "$R" ]; then
-        echo $R
-    else
+      fi
+      R=$(awk -F"\t" -v cmd="$1" "$MSG" /usr/share/clear/commandlist.csv)
+      if [ -n "$R" ]; then
+        echo "$R"
+      else
         MSG='$1 == cmd { print "The command " $1 " is not available, consider using: " $2; exit }'
         awk -F"\t" -v cmd="$1" "$MSG" /usr/share/clear/alternatives.csv
-    fi
-    return 127
-  }
-
-  if [ "$ZSH_VERSION" ]; then
-    command_not_found_handler () {
-      command_not_found_handle "$@"
-      return $?
+      fi
+      return 127
     }
+
+    if [ "$REALNAME" = zsh ]; then
+      command_not_found_handler () {
+        command_not_found_handle "$@"
+        return $?
+      }
+    fi
   fi
 fi
-- 
2.23.0

